src/rapids_dependency_file_generator/schema.json

```diff
@@ -0,0 +1,160 @@
+ {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/rapidsai/dependency-file-generator/v1.0.0/src/rapids_dependency_file_generator/schema.json",
+    "type": "object",
+    "title": "RAPIDS Package Dependency Specification Format",
+    "description": "Consolidated specification of RAPIDS project dependencies.",
+    "properties": {
+        "files": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "object",
+                    "properties": {
+                        "output": {"$ref": "#/$defs/Outputs"},
+                        "includes": {"type": "array", "items": {"type": "string"}},
+                        "matrix": {"$ref": "#/$defs/Matrix"},
+                        "requirements_dir": {"type": "string"},
+                        "conda_dir": {"type": "string"}
+                    },
+                    "additionalProperties": false,
+                    "required": ["output", "includes"]
+                }
+            }
+        },
+        "dependencies": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "object",
+                    "properties": {
+                        "common": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "output_types": {"$ref": "#/$defs/Outputs"},
+                                    "packages": {"$ref": "#/$defs/Packages"}
+                                },
+                                "required": ["output_types", "packages"],
+                                "additionalProperties": false
+                            }
+                        },
+                        "specific": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "output_types": {"$ref": "#/$defs/Outputs"},
+                                    "matrices": {"$ref": "#/$defs/Matrices"}
+                                },
+                                "required": ["output_types", "matrices"],
+                                "additionalProperties": false
+                            }
+                        }
+                    },
+                    "minProperties": 1,
+                    "additionalProperties": false
+                }
+            }
+        },
+        "channels": {"$ref": "#/$defs/Channels"}
+    },
+    "required": ["files", "dependencies"],
+    "additionalProperties": false,
+    "$defs": {
+        "Channel": {
+            "type": "string",
+            "format": "iri-reference"
```

Comment:
Is this restriction on valid channel names documented somewhere? I'm not familiar with it (it certainly seems reasonable, I just don't know).

Reply:
I did a course search and did not find anything definitive, however they are described as URLs here: https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/channels.html#what-is-a-conda-channel .

I assume that IRIs are actually supported even though the documentation speaks of URLs. In this context I would advise to lean towards *less restrictive* so that this tool does not actually limit function. I assume that IRIs are supported even though the documentation speaks of URLs.

Reply:
> In this context I would advise to lean towards less restrictive so that this tool does not actually limit function

That's a reasonable approach. Let's leave it, we can always tighten the restriction later if we discover that IRIs actually pose a problem.